octobot_tentacles_manager/util/hashing.py

```diff
@@ -0,0 +1,40 @@
+#  Drakkar-Software OctoBot-Commons
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+import json
+import inspect
+import hashlib
+
+import octobot_tentacles_manager.configuration.tentacle_configuration as tentacle_configuration
+
+
+def get_tentacles_code_hash(tentacles: list) -> str:
+    full_code = ""
+    for linked_tentacle in tentacles:
+        code_location = linked_tentacle.get_script() if hasattr(linked_tentacle, "get_script") \
+            else linked_tentacle.__class__
+        tentacle_code = inspect.getsource(code_location).replace(" ", "").replace("\n", "")
```

Comment:
Why these replacements?

Reply:
The original idea is to avoid making a new hash when only a space / newline was added as it's not changing the code.
However I'm wondering if we really should do it, the "risk" of people adding a random space is quite low and it might create conflicts with real need spaces